snyk/cli
# Feat/snyk timeout secs

## Pull Request Submission

Please check the boxes once done.

The pull request must:

- **Reviewer Documentation**
- **User facing Documentation**
- **Testing**

## Pull Request Review

All pull requests must undergo a thorough review process before being merged. 
The review process of the code PR should include code review, testing, and any necessary feedback or revisions. 
Pull request reviews of functionality developed in other teams only review the given documentation and test reports. 

Manual testing will not be performed by the reviewing team, and is the responsibility of the author of the PR.

For Node projects: It’s important to make sure changes in `package.json` are also affecting `package-lock.json` correctly.

****************************If a dependency is not necessary, don’t add it.**************************** 

When adding a new package as a dependency, make sure that the change is absolutely necessary. We would like to refrain from adding new dependencies when possible.
Documentation PRs in gitbook are reviewed by Snyk's content team. They will also advise on the best phrasing and structuring if needed.

## Pull Request Approval

Once a pull request has been reviewed and all necessary revisions have been made, it is approved for merging into 
the main codebase. The merging of the code PR is performed by the code owners, the merging of the documentation PR 
by our content writers.


## What does this PR do?

Terminate CLI with exit code EX_UNAVAILABLE after a timeout, when set
using SNYK_TIMEOUT_SECS=N environment variable, where N is the number of
seconds the user is willing to wait for the command to complete.

## Where should the reviewer start?

`setTimeout` function added to `main.go`

## How should this be manually tested?

Build the CLI with `make clean build`. Test it with:

`SNYK_TIMEOUT_SECS=1 ./binary-releases/snyk-macos-arm64 test`

1 second is usually not long enough for `snyk test`, so you should see the CLI terminate with exit code 69 (EX_UNAVAILABLE).

## Any background context you want to provide?

Some CI/CD use cases run Snyk test with "best effort". Aborting the command after a maximum duration provides assurance that Snyk does not hang the CI/CD pipeline.

This PR is stacked on #4941 so it may be helpful to review the last commit in isolation.

Still a draft because:
- This needs some kind of test (probably something scripted in Docker)
- Depends on https://github.com/snyk/go-application-framework/pull/112 landing. Until then a `replace` directive is used to build against that dependency's PR branch so you can build and test this in the meantime.

## What are the relevant tickets?


## Screenshots


## Additional questions
